src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,1323 @@
+        descanner.scanGain.subscribe(self._startCalibration)
+
+        # Subscribing to the scanner VA's
+        scanner.dwellTime.subscribe(self._startCalibration)
+        scanner.rotation.subscribe(self._startCalibration)
+        scanner.scanDelay.subscribe(self._startCalibration)
+        scanner.scanOffset.subscribe(self._startCalibration)
+        scanner.scanGain.subscribe(self._startCalibration)
+
+        # Subscribing to the scanner VA
+        mppc.cellCompleteResolution.subscribe(self._startCalibration)
+
+        # Assemble the calibration data and retrieve input values
+        self._calibrationParameters = self._assembleCalibrationData()
+        self.asmApiPostCall("/scan/start_calibration_loop", 204, data=self._calibrationParameters.to_dict())
+
+    def _assembleCalibrationData(self):
+        """
+        Assemble the calibration data and retrieve the input values from the scanner, descanner and mppc VA's.
+
+        :return calibration_data (CalibrationLoopParameters object): Object holding the calibration data which can be
+        send to the ASM API
+        """
+        descanner = self._mirror_descanner
+        scanner = self._ebeam_scanner
+        mppc = self._mppc
+
+        descan_rotation = descanner.rotation.value
+        descan_offset = descanner.scanOffset.value
+        dwell_time = scanner.dwellTime.value
+        dwell_time_ticks = scanner.getTicksDwellTime()
+        scan_rotation = scanner.rotation.value
+        scan_delay = scanner.getTicksScanDelay()
+        scan_offset = scanner.scanOffset.value
+        resolution = mppc.cellCompleteResolution.value[0]
+
+        # Check if the descanner clockperiod is still a multiple of de scanner clock period, otherwise raise error.
+        if not almost_equal(descanner.clockPeriod.value % scanner.clockPeriod.value, 0):
+            logging.error("Descanner and/or scanner clock period changed which means the descanner period is no "
+                          "longer a whole multiple of scanner clock periods. This means that calibration is no "
+                          "longer accurate.")
+            raise ValueError("Descanner clock period is no longer a whole multiple of the scanner clock period.")
+
+        # Creation of setpoints
+        flyback_time = descanner.physicalFlybackTime
+        remainder_scanning_time = (dwell_time * resolution) % descanner.clockPeriod.value
+        if remainder_scanning_time is not 0:
+            # Add adjusted flyback time if there is a remainder of scanning time by adding one setpoint to ensure the
+            # total scanning period is equal to a whole number of descan clock periods.
+            flyback_time = flyback_time + descanner.clockPeriod.value
+
+        # Period of the calibration signal in seconds
+        calibration_period = (dwell_time * resolution) + flyback_time
+
+        # Support a calibration freq from 125 to 25000 Hz
+        if not scanner.dwellTime.range[0] < calibration_period < 0.008:
```

Comment:
You told me last week to change the range of calibration frequencies we supported to be up to the minimum of one dwell time period. Because you told me I should implement that, before the max calibration frequency supported was 5000

Reply:
The minimum allowed for the calibration dwell time is the minimum allowed for the dwell time (400ns). Here, you check that the whole line scan time (=calibration_period) is at least 400ns. That doesn't make sense to me. We can solve this offline if you want.